config/initializers/s3.rb

```diff
@@ -6,9 +6,11 @@
 AWS_SECRET_ACCESS_KEY = ENV['AWS_SECRET_ACCESS_KEY']
 AWS_BUCKET            = ENV['AWS_BUCKET']
 AWS_REGION            = ENV['AWS_REGION']
+AWS_ENDPOINT_URL_S3   = ENV['AWS_ENDPOINT_URL_S3']
 
 AWS_CLIENT_OPTIONS = {
   access_key_id: AWS_ACCESS_KEY_ID,
   secret_access_key: AWS_SECRET_ACCESS_KEY,
   region: AWS_REGION,
+  endpoint: AWS_ENDPOINT_URL_S3
```

Comment:
Does this break the default case where `endpoint: nil`? You may need to only set this key the value is non-nil.

Reply:
I didn't test the default configuration, but with specifying it, it works with no issues

Reply:
Would this be enough to support the default config?
```suggestion
  endpoint: AWS_ENDPOINT_URL_S3 or nil
```

Reply:
No, because that does nothing since `AWS_ENDPOINT_URL_S3` is already nil if it's not specified.

Right now, I get the following error when `endpoint: nil` when initializing the client:

```
error: missing required option `:endpoint' (ArgumentError)
```

Meaning, this will break the majority of Keygen installs, since this would be a new option.

This new feature needs to be implemented in a backwards compatible way. The above error indicates that `endpoint:` should only be set when present, since it's not allowed to be `nil`. Easiest way to do that would be to `compact` the `AWS_CLIENT_OPTIONS` hash before freezing it, which would remove any `nil` values.

Reply:
I have addressed this in [`c5a45a3` (#895)](https://github.com/keygen-sh/keygen-api/pull/895/commits/c5a45a3a41ba2a2461755df85e6e7e1714c434e5)